### PR TITLE
Fix updating keychain entry

### PR DESCRIPTION
**Pull request by armstrongnate**
[ignore-commit-lint]

I noticed that my login sessions were not persisting. Turns out I was getting `errSecItemNotFound` from `SecItemAdd` and then `errSecDuplicateItem` from `SecItemUpdate`, identitical to [this SO post](https://stackoverflow.com/questions/22403734/keychain-item-reported-as-errsecitemnotfound-but-receive-errsecduplicateitem-o).

Turns out that we should only query on `kSecClass`, `kSecAttrService` and `kSecAttrAccount` because additional fields (like `kAttrAccessible`) will _"exclude any existing item with the same unique key, but with different attributes."_

The new approach is taken from [this post](https://forums.developer.apple.com/message/259602#259602) from an Apple developer.

### CODE REVIEW

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

File: Core/Core/AppEnvironment/Keychain.swift [hunk 1]
```diff
@@ -30,33 +30,27 @@ class Keychain {
...
-            return true
+        let exists = SecItemCopyMatching(query as CFDictionary, nil) == noErr
+        var status: OSStatus?
+        if exists {
+            status = SecItemUpdate(query as CFDictionary, [kSecValueData: data] as CFDictionary)
+        } else {
+            query[kSecAttrAccessible] = kSecAttrAccessibleWhenUnlocked
+            query[kSecValueData] = data
+            if accessGroup?.isEmpty == false {
+                query[kSecAttrAccessGroup] = accessGroup
```
Comment (vanwagonet) [inline Core/Core/AppEnvironment/Keychain.swift]: I think we want this to always be part of the query

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1003974&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|0
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Comment by inst-danger** [comment]
Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Canvas iOS** | 94.98% | 94.85% | 0.13%
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Pull request merged**